src/odemis/dataio/catmaid.py

```diff
@@ -0,0 +1,351 @@
+        FILE_BASED_WITH_ZOOM_DIRS: '{image_base}{depth}/{zoom_level}/{row}_{col}.{file_extension}',
+        DIR_BASED: '{image_base}{zoom_level}/{depth}/{row}/{col}.{file_extension}',
+        RENDER_SERVICE: '{image_base}largeDataTileSource/{tile_width}/{tile_height}/{zoom_level}/'
+                        '{depth}/{row}/{col}.{file_extension}',
+        FLIXSERVER: '{image_base}{depth}/{row}_{col}_{zoom_level}.{file_extension}',
+    }
+    try:
+        tile_url = tile_urls[tile_source_type]
+    except KeyError:
+        logging.info("{} is not supported by TileFetcher, supported types are below:\n\t{}".format(
+            tile_source_type, '\n\t'.join(str(k) for k in sorted(tile_urls))))
+        return image_base
+    return tile_url.format(image_base=image_base, zoom_level=zoom_level, depth=depth, row=row, col=col,
+                           file_extension=file_extension, tile_width=tile_width, tile_height=tile_height)
+
+
+def response_to_array(response):
+    """
+    response (Response): http response for the requested image.
+    return:
+       image (numpy array): the requested image from the response.
+    """
+    response.raise_for_status()
+    content_type = response.headers['Content-Type']
+
+    if content_type in SUPPORTED_CONTENT_TYPES:
+        buffer = BytesIO(response.content)  # opening directly from raw response doesn't work for JPEGs
+        raw_img = Image.open(buffer)
+        return numpy.array(raw_img)
+    else:
+        logging.info('Image fetching is only implemented for greyscale PNG and JPEG, not {}'.format(
+            content_type.upper().split('/')[1]
+        ))
+
+
+def get_stack_info(base_url, project_id, stack_id):
+    """Get the stack info for a project hosted at the base url."""
+    token, _, _ = read_config_file(base_url, token=True)
+    stack_url = "{base_url}/{project_id}/stack/{stack_id}/info"
+    stack_url = stack_url.format(base_url=base_url, project_id=project_id, stack_id=stack_id)
+    response = requests.get(stack_url, auth=CatmaidApiTokenAuth(token)) if token else requests.get(stack_url)
+    # A PermissionError is raised when authentication fails.
+    if response.status_code == 404:
+        raise ValueError("no stack info at this url, check the base url {base_url}".format(base_url=base_url))
+    elif "PermissionError" in requests.get(stack_url, auth=CatmaidApiTokenAuth("asdf")).text:
+        raise ValueError("Authentication Failure")
+    elif "DoesNotExist" in response.text:
+        # Catmaid returns a 200 code with an error text field, when there's no stack info at that url.
+        raise ValueError(
+            "No stack info at this url {stack_url}, check project id {project_id} and stack id {stack_id}".format(
+                stack_url=stack_url, project_id=project_id, stack_id=stack_id))
+    stack_info = response.json()
+    # resolution is given in nanometers by catmaid.
+    stack_info["resolution"] = {k: v * 1e-9 for k, v in stack_info["resolution"].items()}
```

Comment:
Please leave the stack_info as-is. The caller of this function should take care of reading it properly (ie, multiplying by 1e-9 when needed).